### PR TITLE
Fixes win32 input for Danish keyboard layout

### PR DESCRIPTION
**Pull request by R-Michelsen**
I've made a small change to the input handling to match that of neovim-qt which handles characters "[", "]", "@" and "\\" properly on my Danish keyboard layout which neovide didn't. 

Since I am largely unfamiliar with vim and its input model, I don't know if this breaks other layouts but I would assume not since neovim-qt uses this logic. In any case I thought I would open this PR in case you want to merge it.

The change is taken from [input_win32.cpp](https://github.com/equalsraf/neovim-qt/blob/master/src/gui/input_win32.cpp) lines 28-37 and is a minor change from what already existed in the codebase.

### CODE REVIEW

**Comment by R-Michelsen** [comment]
I just noticed this seems to mess up clipboard functionality, (as is messed up in neovim-qt currently aswell) I will try to fix this then reopen the PR

**Comment by R-Michelsen** [comment]
Actually, after more investigation believe the clipboard "issues" are present even in the master branch so I'll reopen the PR. I'm not sure they are issues really though, it may just be that I expect the app to paste the clipboard contents on CTRL+V but that might be easier to handle through vim mappings.

File: src/bridge/layouts/mod.rs [hunk 1]
```diff
@@ -64,20 +64,20 @@ fn append_modifiers(
...
+	if ctrl && !alt {
         special = true;
         result = format!("C-{}", result);
     }
-    if alt {
+    if alt && !ctrl {
         special = true;
         result = format!("M-{}", result);
-    }
+	}
```
Comment (Kethku) [inline src/bridge/layouts/mod.rs:80]: This breaks C-M-{} bindings on keyboard layouts which allow such bindings

Comment (R-Michelsen) [reply src/bridge/layouts/mod.rs:80]: Hmm I see, thats interesting. I'll investigate further, as I noted I more or less stole this change from neovim-qt but perhaps they don't handle C-M-{} bindings either? In any case I'll have to further investigate why `[ ] @ \ ` and a few other keys weren't working for me in the first place.

Comment (Kethku) [reply src/bridge/layouts/mod.rs:80]: Interesting. Its very possible that I don't understand keyboard input in neovim yet either :(

**Comment by R-Michelsen** [comment]
After some consideration, I think the best workaround for me is to fix the broken Alt+Gr keys in my init.vim like this:
```vim
" Fix broken Alt+Gr characters
inoremap <C-M-@> @
inoremap <C-M-£> £
inoremap <C-M-]> ]
inoremap <C-M-[> [
inoremap <C-M-\> \
```
Ultimately from what I can tell each editor has its own implementation that accommodates some set of keyboard layouts but probably not all. Therefore it's probably best to leave the implementation as is. Feel free to include the work-around somewhere in your docs if you want, but it doesn't seem like a frequent issue, maybe we aren't many Danish vim users or something 😃

**Comment by Kethku** [comment]
Unfortunately I don't know what the "correct" solution is :( Its a hard problem which is made harder by the windowing system.

**Pull request closed without merging**